_includes/reactions.njk

```diff
@@ -0,0 +1,36 @@
+{#
+  Make sure not to include this more than once into a page
+  it is right now tied to the main layout, so it will already
+  be included in every page that uses that
+#}
+
+<div id="reactions-list">
+  <span id="big-smile" class="reaction" data-href="/reactions/big-smile/">😀</span>
+  <span id="smile" class="reaction" data-href="/reactions/smile/">🙂</span>
+  <span id="neutral" class="reaction" data-href="/reactions/neutral/">😑</span>
+  <span id="frown" class="reaction" data-href="/reactions/frown/">🙁</span>
+  <span id="angry" class="reaction" data-href="/reactions/angry/">😡</span>
+</div>
+
+<script type="application/javascript">
+  const reactions = document.querySelectorAll(".reaction")
+
+  let reactionListener = (e) => {
+    // If we already removed it, just return
+    if (!e.target.dataset.href) return;
+
+    // Make a call to the url in the background
+    fetch(e.target.dataset.href)
+      .then(() => {
+        // Disable the links
+        for (const r of reactions) {
+          r.removeAttribute("data-href")
+          r === e.target ? r.classList.add("clicked") : r.classList.add("notclicked")
+        }
+      })
+  }
+
+  for (const reaction of reactions) {
```

Comment:
Re: Whether or not there's any benefit other than it looking better/having less code: No.

But you can say the same about why you'd use a for loop instead of:

```
var i = 0;
while (i < reactions.length) {
  reactions[i].addEventListener('click', reactionListener);
  i = i + 1;
}
```

Reply:
cool, i just wasn't sure if there was some subtle javascript performance weirdness that i wasn't understanding 😄